frame/support/src/storage/mod.rs

```diff
@@ -44,71 +44,91 @@ pub mod types;
 pub mod unhashed;
 pub mod weak_bounded_vec;
 
-#[cfg(all(feature = "std", any(test, debug_assertions)))]
-mod debug_helper {
-	use std::cell::RefCell;
+mod transaction_level_tracker {
+	const TRANSACTION_LAYER_KEY: &'static [u8] = b":transactional_levels:";
+	type Level = u8;
 
-	thread_local! {
-		static TRANSACTION_LEVEL: RefCell<u32> = RefCell::new(0);
+	fn get_transaction_level() -> Level {
+		crate::storage::unhashed::get_or_default::<Level>(TRANSACTION_LAYER_KEY)
 	}
 
-	pub fn require_transaction() {
-		let level = TRANSACTION_LEVEL.with(|v| *v.borrow());
+	fn set_transaction_level(level: &Level) {
+		crate::storage::unhashed::put::<Level>(TRANSACTION_LAYER_KEY, level);
```

Comment:
you tracking storage layer with a storage? won't the commit/revert of underlying storage causing some conflicts with his?

Reply:
i do it before we call `start_transactional`, so it should be okay?

Reply:
I wrote a test which checks this specifically. it works :)